scripts/profiler/rtt2socket.py

```diff
@@ -148,26 +148,17 @@ def disconnect(self):
         self.logger.info("Disconnected from device")
 
     def read_bytes(self):
-        bufs = bytearray()
-        while True:
-            try:
-                buf = self.jlink.rtt_read(self.rtt_up_channels['data'],
-                                          self.config['rtt_read_chunk_size'],
-                                          encoding=None)
-            except APIError:
-                self.logger.error("Problem with reading RTT data")
-                self.disconnect()
-                sys.exit()
-
-            if len(buf) > 0:
-                bufs.extend(buf)
-
-            if len(buf) < self.config['rtt_additional_read_thresh']:
-                break
+        try:
+            buf = self.jlink.rtt_read(self.rtt_up_channels['data'],
+                                      self.config['rtt_read_chunk_size'],
+                                      encoding=None)
 
-            time.sleep(0.05)
+        except APIError:
+            self.logger.error("Problem with reading RTT data")
+            self.disconnect()
+            sys.exit()
 
-        return bufs
+        return buf
```

Comment:
I think we shouldn't send empty buffer over sockets. It just adds unnecessary traffic flow. Actually this issue is present since this PR: https://github.com/nrfconnect/sdk-nrf/pull/5837.